pkg/shp/reactor/pod_watcher.go

```diff
@@ -120,15 +120,19 @@ func (p *PodWatcher) handleEvent(pod *corev1.Pod, event watch.Event) error {
 	return nil
 }
 
-// Start runs the event loop based on a watch instantiated against informed pod. In case of errors
-// the loop is interrupted.
-func (p *PodWatcher) Start(listOpts metav1.ListOptions) (*corev1.Pod, error) {
+func (p *PodWatcher) Connect(listOpts metav1.ListOptions) error {
```

Comment:
Would be great to add a doc-comment explaining how the connect helps the multi-threading, etc.